apps/wallet/src/ui/app/hooks/useGetTransactionsByAddress.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiAddress, type SuiTransactionResponse } from '@mysten/sui.js';
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+
+import { api } from '_redux/store/thunk-extras';
+const STALE_TIME = 1000 * 2;
+
+async function getTransactionsByAddress(
+    address: string,
+    cursor?: string
+): Promise<SuiTransactionResponse[]> {
+    const rpc = api.instance.fullNode;
+    const txnsIds = await api.instance.fullNode.getTransactions(
+        {
+            FromAddress: address,
+        },
+        cursor
+    );
+    return rpc.getTransactionWithEffectsBatch(txnsIds.data);
+}
+
+// Fetch transactions on mount and every 2 seconds
```

Comment:
This isn't really accurate, it just is considered stale after 2 seconds, but that doesn't refetch. Honestly with 2 second stale time you might as well keep it at 0 (or increase it to something meaningful).

Reply:
done